_storage/gaia-admin.md

```diff
@@ -0,0 +1,338 @@
+    "timestamp": true,
+    "stringify": true,
+    "colorize": true,
+    "json": true
+  },
+  "port": 8009,
+  "apiKeys": [ "hello" ],
+  "gaiaSettings": {
+    "configPath": "/tmp/hub-config/config.json"
+  },
+  "reloadSettings": {
+    "command": "/bin/sh",
+    "argv": [
+      "-c",
+      "docker restart docker_hub_1 &"
+    ],
+    "env": {},
+    "setuid": 1000,
+    "setgid": 1000
+  }
+}
+```
+
+The `port` is the port where Gaia is running. The `apiKeys` field is key used for making calls to the hub.  The `gaiaSettings`
+
+
+The `argsTransport` section configures the hub logging. The service uses the `winston` logging service. Refer to their documentation for full details on the [logging configuration options](https://github.com/winstonjs/winston).
+
+<table class="uk-table uk-table-small uk-table-divider">
+   <thead>
+      <tr>
+         <th>Field</th>
+         <th>Description</th>
+      </tr>
+   </thead>
+   <tbody>
+      <tr>
+         <td><code>level</code></td>
+         <td>Lowest level this transport will log. (default: <code>info</code>)</td>
+      </tr>
+      <tr>
+         <td><code>handleException</code></td>
+         <td>Set to true to have this transport handle exceptions. (default: <code>false</code>)</td>
+      </tr>
+      <tr>
+         <td><code>timestamp</code></td>
+         <td>The timestamp when the message was received.</td>
+      </tr>
+      <tr>
+         <td><code>stringify</code></td>
+         <td>Converts the output to a JSON string.</td>
+      </tr>
+      <tr>
+         <td><code>colorize</code></td>
+         <td>Colorizes the standard logging level</td>
+      </tr>
+      <tr>
+         <td><code>json</code></td>
+         <td>Log format.</td>
+      </tr>
+   </tbody>
+</table>
+
+The `reloadSettings` configure the command that is used to reload your Gaia hub.
+
+<table class="uk-table uk-table-small uk-table-divider">
+   <thead>
+      <tr>
+         <th>Field</th>
+         <th>Description</th>
+      </tr>
+   </thead>
+   <tbody>
+      <tr>
+         <td><code>command</code></td>
+         <td>A command which reloads the Gaia hub service.</td>
+      </tr>
+      <tr>
+         <td><code>argv</code></td>
+         <td>An array containing the command arguments.</td>
+      </tr>
+      <tr>
+         <td><code>env</code></td>
+         <td>This is a key/value list of any environment variables
+           that need to be set for the command to run. This is optional.</td>
+      </tr>
+      <tr>
+         <td><code>setuid</code></td>
+         <td>This is the UID under which the command will be run. This is optional.</td>
+      </tr>
+      <tr>
+         <td><code>setgid</code></td>
+         <td>This is the GID under which the command will run. This is optional.</td>
+      </tr>
+   </tbody>
+</table>
+
+
+## Using the admin service APIs
+
+You use the admin service APIs to manage the hub. Get calls require that you pass the `API_KEY` set in the admin service config:
```

Comment:
We should add for people who set up their hub service via the AMI:

If you set up your hub service via the AMI, the admin interface is only accessible from within your virtual machine. To execute admin functions on your gaia hub, you'll need to ssh into your virtual machine:

```
ssh -t -i <your keyfile.pem> -A core@<public ip address>
```

Reply:
@wileyj and I had a conversation about this.  He said it was possible to make the curl calls from my local machine (after all just a curl).  If I do do that, I believe I need to set up local variables or use the AWS CLI to set up.  I've done this with similar environments in the past.

Reply:
Hmm -- I think you'd need to set up an ssh tunnel to make this work. As currently configured in the AMI, requests to `/admin` are blocked for everything but the localhost, and the port `8009` isn't configured to be open.

You could do something like this:

```
ssh -t -i <keyfile.pem> -L 8009:localhost:8009 -A core@<public ip address>
```

Which starts an SSH tunnel. Then you'd be able to use the curl commands from your local machine.

Reply:
fixed per meeting convo